ui-admin/src/forms/designer/NewQuestionForm.test.tsx

```diff
@@ -28,4 +29,35 @@ describe('NewQuestionForm', () => {
     expect((questionTypeSelect as HTMLSelectElement).value).toBe('checkbox')
     expect(screen.getByText(questionTypeDescriptions.checkbox)).toBeInTheDocument()
   })
+
+  test('renders freetext input', () => {
+    //Arrange
+    render(<NewQuestionForm onCreate={() => jest.fn()} readOnly={false} />)
+
+    //Act
+    const questionTypeSelect = screen.getByLabelText('Question type')
+    fireEvent.change(questionTypeSelect, { target: { value: 'freetext' } })
+
+    //Assert
+    const freetextInput = screen.getByLabelText('Freetext')
+    expect((questionTypeSelect as HTMLSelectElement).value).toBe('freetext')
+    expect(freetextInput as HTMLInputElement).toBeInTheDocument()
+    expect(screen.getByText(questionTypeDescriptions.text)).toBeInTheDocument() //initial question type is text
+  })
+
+  test('updates to the appropriate QuestionDesigner based on freetext input', async () => {
+    //Arrange
+    render(<NewQuestionForm onCreate={() => jest.fn()} readOnly={false}/>)
+
+    //Act
+    const questionTypeSelect = screen.getByLabelText('Question type')
+    fireEvent.change(questionTypeSelect, { target: { value: 'freetext' } })
+    const freetextInput = screen.getByLabelText('Freetext') as HTMLInputElement
+    userEvent.pointer({ target: freetextInput, offset: 0, keys: '[MouseLeft]' })
+    userEvent.keyboard('This is a question!\nThis is an option!\nThis is another option!')
```

Comment:
Most `userEvent` methods return Promises and should be awaited

```suggestion
    await userEvent.pointer({ target: freetextInput, offset: 0, keys: '[MouseLeft]' })
    await userEvent.keyboard('This is a question!\nThis is an option!\nThis is another option!')
```

https://testing-library.com/docs/user-event/pointer
https://testing-library.com/docs/user-event/keyboard

Awaiting these methods may allow getting rid of the `waitFor`s below.

Also wondering if this combination of `pointer` and `keyboard` could be replaced by [`type`](https://testing-library.com/docs/user-event/utility#type).